lib/spack/spack/test/util/prefix.py

```diff
@@ -36,6 +36,13 @@ def test_prefix_attributes():
     assert prefix.include == '/usr/include'
 
 
+def test_prefix_join():
+    """Test prefix join  ``prefix.join(...)``"""
+    prefix = Prefix('/usr')
+
+    assert prefix.join('a_{0}'.format(1)).lib64 == '/usr/a_1/lib64'
```

Comment:
I would add tests for other more common use cases (dashes or periods in filename). I know it won't make a difference, but that's how it will most commonly be used.

Reply:
I added a couple of more. Let me know if you want anything else to be tested.

Reply:
Oh, maybe do some chained examples, like:
```python
prefix.bin.join('executable.sh')
prefix.share.join('pkg-config').join('foo.pc')
prefix.join('dashed-directory').foo
```
Also assert that the result is of type `Prefix`, not type `str`.

Reply:
will do!

Reply:
@adamjstewart I added those tests.

Reply:
If you save them in variables you won't have to duplicate so much.